src/forms/FormControl/FormControl.test.tsx

```diff
@@ -0,0 +1,181 @@
+  const mockTestId = 'my-test-id'
+
+  afterEach(cleanup)
+
+  it('renders a text input form control correctly into the document', () => {
+    const {getByRole, getByLabelText} = render(
+      <FormControl>
+        <FormControl.Label>{mockFormControlLabel}</FormControl.Label>
+        <TextInput />
+      </FormControl>
+    )
+
+    const labelEl = getByLabelText(mockFormControlLabel)
+    const inputEl = getByRole('textbox')
+
+    expect(labelEl).toBeInTheDocument()
+    expect(inputEl).toBeInTheDocument()
+  })
+
+  it('has no a11y violations', async () => {
+    const {container} = render(
+      <FormControl>
+        <FormControl.Label>{mockFormControlLabel}</FormControl.Label>
+        <TextInput />
+      </FormControl>
+    )
+    const results = await axe(container)
+
+    expect(results).toHaveNoViolations()
+  })
+
+  it('generates a unique ID and forwards to relevant children', async () => {
+    const {container, getByRole, getByTestId} = render(
+      <FormControl id={mockFormControlId} data-testid={mockTestId}>
+        <FormControl.Label>{mockFormControlLabel}</FormControl.Label>
+        <TextInput />
+      </FormControl>
+    )
+    const labelEl = container.querySelector('label')
+    const labelForValue = labelEl?.getAttribute('for')
+
+    const inputEl = getByRole('textbox')
+    const rootEl = getByTestId(mockTestId)
+
+    const inputIdValue = inputEl.getAttribute('id')
+    const inputNameValue = inputEl.getAttribute('name')
+
+    expect(inputIdValue).toEqual(labelForValue)
+    expect(inputNameValue).toEqual(labelForValue)
+    expect(rootEl.getAttribute('id')).toBe(mockFormControlId)
+  })
+
+  it('can forward an ID override to relevant children', async () => {
+    const {container, getByTestId, getByRole} = render(
+      <FormControl id={mockFormControlId} data-testid={mockTestId}>
+        <FormControl.Label>{mockFormControlLabel}</FormControl.Label>
+        <TextInput />
+      </FormControl>
+    )
+
+    const labelEl = container.querySelector('label')
+    const labelForValue = labelEl?.getAttribute('for')
+
+    const inputEl = getByRole('textbox')
+    const rootEl = getByTestId(mockTestId)
+
+    const inputIdValue = inputEl.getAttribute('id')
+    const inputNameValue = inputEl.getAttribute('name')
+
+    expect(inputIdValue).toEqual(labelForValue)
+    expect(inputNameValue).toEqual(labelForValue)
+    expect(rootEl.getAttribute('id')).toBe(mockFormControlId)
+  })
+
+  it('applies error state to the inputs and form validation', async () => {
+    const expectedMessage = 'My error message'
+
+    const {container, getByText, getByRole} = render(
+      <FormControl validationStatus="error">
+        <FormControl.Label>{mockFormControlLabel}</FormControl.Label>
+        <TextInput />
+        <FormControl.Validation>{expectedMessage}</FormControl.Validation>
+      </FormControl>
+    )
+
+    const errorIcon = container.querySelector('svg.octicon-alert-fill')
```

Comment:
This is referencing the 12px icon but is being rendered at 16px. We should make an icon request to add and `alert-fill` 16px icon here since we are intending for larger validation messages.

Reply:
Sorry @ashygee, not quite following on this one. Do you mean you want me to add a 16px SVG to Octicons and then pull it in here? 

As it's a vector it will scale without distortion, so is that necessary? I've already bumped the validation text size in an earlier commit.

![Screenshot 2022-09-07 at 16 21 03](https://user-images.githubusercontent.com/13340707/188917184-6202c945-b528-4c69-8002-550dd14b2424.png)